ffigen/entrypoint.h

```diff
@@ -0,0 +1,22 @@
+#include <stdint.h>
+
+// In order to cheat ffigen and hide the dart-dl headers, we define the dart-api.h include guard here,
+// which prevents libclang from processing the header file at all
+#define RUNTIME_INCLUDE_DART_API_H_
+
+// however we still need these types from dart-api.h in our headers
+typedef struct _Dart_Handle* Dart_Handle;
+typedef Dart_Handle Dart_PersistentHandle;
+typedef struct _Dart_WeakPersistentHandle* Dart_WeakPersistentHandle;
+typedef struct _Dart_FinalizableHandle* Dart_FinalizableHandle;
+typedef int64_t Dart_Port;
+
+// The C API header
+#include <realm.h>
+
+// List all the headers that should be processed by ffigen, relative to the src folder
+#include <realm_dart_scheduler.h>
+#include <realm_dart.h>
```

Comment:
What good is this for if we do the same as the `config.yaml`

Reply:
I suggest we continue with the `config.yaml` much simpler than fiddling with a workaround headers.

Reply:
we have them regardless in other places. I suggest we continue using it.

Reply:
We should stop using symlinks in other places as well 😛
But please go over the PR again when you have more time - the single entry point header is necessary both to get rid of symlinks, but to also work around a breaking change in ffigen 5 when it comes to encountering the same declaration multiple times.